app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person_data in people:
+        person = Person(person_data["name"], person_data["age"])
+        person_list.append(person)
+
+    for person_data in people:
+        person = Person.people[person_data["name"]]
+        if "wife" in person_data and person_data["wife"]:
+            person.wife = Person.people[person_data["wife"]]
+        elif "husband" in person_data and person_data["husband"]:
+            person.husband = Person.people[person_data["husband"]]
```

Comment:
This part of the code can be simplified and made more readable. You are iterating over the `people` list twice, but it can be done in a single pass. Also, you can avoid nested `if` statements by using the `or` logical operator. Here's a hint: create a Person instance and assign a wife or husband (if they exist) in the same loop.